custom/icds_reports/utils/aggregation_helpers/distributed/agg_gov_dashboard.py

```diff
@@ -1,60 +1,41 @@
-from dateutil.relativedelta import relativedelta
-
 from custom.icds_reports.const import AGG_GOV_DASHBOARD_TABLE
 from custom.icds_reports.utils.aggregation_helpers import transform_day_to_month, month_formatter
-from custom.icds_reports.utils.aggregation_helpers.distributed.base import BaseICDSAggregationDistributedHelper
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import AggregationPartitionedHelper
 
 
-class AggGovDashboardHelper(BaseICDSAggregationDistributedHelper):
+class AggGovDashboardHelper(AggregationPartitionedHelper):
     helper_key = 'agg-gov-dashboard'
     base_tablename = AGG_GOV_DASHBOARD_TABLE
+    staging_tablename = 'staging_{}'.format(AGG_GOV_DASHBOARD_TABLE)
 
-    def __init__(self, month):
-        self.month_start = transform_day_to_month(month)
-        self.prev_month_start = self.month_start - relativedelta(months=1)
-        self.next_month_start = self.month_start + relativedelta(months=1)
-
-    def aggregate(self, cursor):
-        drop_table_query = self.drop_table_if_exists()
-        cursor.execute(drop_table_query)
-
-        create_table_query, create_params = self.create_child_table()
-        cursor.execute(create_table_query, create_params)
-
-        agg_query, agg_params = self.aggregate_query()
-
-        cursor.execute(agg_query, agg_params)
-
-        update_queries = self.update_queries()
+    @property
+    def monthly_tablename(self):
+        return "{}_{}".format(self.base_tablename, self.month.strftime("%Y-%m-%d"))
 
-        for query, param in update_queries:
-            cursor.execute(query, param)
+    @property
+    def previous_agg_table_name(self):
+        return f"previous_{self.monthly_tablename}"
 
-        index_query = self.indexes()
-        cursor.execute(index_query)
+    @property
+    def model(self):
+        from custom.icds_reports.models.aggregate import AggGovernanceDashboard
+        return AggGovernanceDashboard
 
     @property
-    def tablename(self):
-        return "{}_{}".format(self.base_tablename, self.month_start.strftime("%Y-%m-%d"))
+    def temporary_tablename(self):
+        return 'tmp_{}'.format(self.base_tablename)
 
-    def drop_table_if_exists(self):
-        return """
-        DROP TABLE IF EXISTS "{table_name}"
-        """.format(table_name=self.tablename)
+    def create_temporary_table(self):
+        return f"""
+        CREATE UNLOGGED TABLE "{self.temporary_tablename}" (LIKE "{self.base_tablename}" INCLUDING INDEXES);
```

Comment:
i dont think you want unlogged here. this will prevent the table from being replicated to the standbys